### PR TITLE
Reduce the clusterable object UBO size below 16384 for WebGL 2.

### LINKED ISSUE #15998: webgl2: `3d_scene` example panics
## Bevy version

main
François bisected to #13497 previously

## Relevant system information

AdapterInfo { name: "ANGLE (Apple, ANGLE Metal Renderer: Apple M1 Max, Unspecified Version)", vendor: 4203, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "WebGL 2.0 (OpenGL ES 3.0 Chromium)", backend: Gl }
Chrome Version 129.0.6668.101 (Official Build) (arm64)

## What you did

```
cargo run -p build-wasm-example -- --api webgl2 3d_scene
basic-http-server examples/wasm/
```

## What went wrong

```
wasm_example.js:1951 panicked at /Users/me/.cargo/registry/src/index.crates.io-6f17d22bba15001f/wgpu-22.1.0/src/backend/wgpu_core.rs:3411:5:
wgpu error: Validation Error

Caused by:
  In Device::create_bind_group, label = 'mesh_view_bind_group'
    Buffer binding 8 range 20480 exceeds `max_*_buffer_binding_size` limit 16384
```

## Additional information

This seems to be platform specific.

I tested on:

AdapterInfo { name: "ANGLE (NVIDIA, NVIDIA GeForce RTX 4080 SUPER (0x00002702) Direct3D11 vs_5_0 ps_5_0, D3D11)", vendor: 4318, device: 0, device_type: Other, driver: "", driver_info: "WebGL 2.0 (OpenGL ES 3.0 Chromium)", backend: Gl }
Chrome 129

And the example worked fine.

### Previous discussion

> from https://docs.rs/wgpu/latest/wgpu/struct.Limits.html#method.downlevel_webgl2_defaults, I guess max_*_buffer_binding_size is max_uniform_buffer_binding_size which should be limited to 16KiB on WebGL2. It's not mandatory for browser to respect it, but it's the standard... and we're requesting more
> - françois

### PR DESCRIPTION
**Pull request by pcwalton**
The PCSS PR #13497 increased the size of clusterable objects from 64 bytes to 80 bytes but didn't decrease the UBO size to compensate, so we blew past the 16kB limit on WebGL 2. This commit fixes the issue by lowering the maximum number of clusterable objects to 204, which puts us under the 16kB limit again.

Closes #15998.

### CODE REVIEW

**Review by tychedelia — APPROVED** [review]
would be nice if we could catch this at compile time

File: crates/bevy_pbr/src/cluster/mod.rs [hunk 1]
```diff
@@ -34,7 +34,7 @@ mod assign;
 mod test;
 
 // NOTE: this must be kept in sync with the same constants in pbr.frag
-pub const MAX_UNIFORM_BUFFER_CLUSTERABLE_OBJECTS: usize = 256;
+pub const MAX_UNIFORM_BUFFER_CLUSTERABLE_OBJECTS: usize = 204;
```
Comment (mockersf) [inline crates/bevy_pbr/src/cluster/mod.rs:37]:
```suggestion
pub const MAX_UNIFORM_BUFFER_CLUSTERABLE_OBJECTS: usize = 204;
// A buffer of MAX_UNIFORM_BUFFER_CLUSTERABLE_OBJECTS GpuClusterableObject items must hold in 16384 bytes
const _: () =
    assert!(size_of::<GpuClusterableObject>() * MAX_UNIFORM_BUFFER_CLUSTERABLE_OBJECTS <= 16384);
```

let's check it at compile time

**Comment by JMS55** [comment]
Might make sense to revisit the cluster data design at some point, it's getting kinda fat, and we're planning on adding decals too.

Is 204 correct though? 16000 / 80 = 200.

File: crates/bevy_pbr/src/cluster/mod.rs [hunk 2]
```diff
@@ -34,7 +34,7 @@ mod assign;
 mod test;
 
 // NOTE: this must be kept in sync with the same constants in pbr.frag
```
Comment (JMS55) [inline crates/bevy_pbr/src/cluster/mod.rs:36]:
```suggestion
// NOTE: this must be kept in sync with the same constants in mesh_view_types.wgsl
```

**Comment by pcwalton** [comment]
@JMS55 16384 / 80 == 204.8.

**Comment by pcwalton** [comment]
@JMS55 This is actually kind of future proof. Padding added 3 extra fields, which should be more than enough space for the decal info.

**Comment by JMS55** [comment]
Ah, KiB not kB 😅

**Review by kristoff3r — APPROVED** [review]

**Review by cart — APPROVED** [review]

File: crates/bevy_pbr/src/cluster/mod.rs [hunk 3]
```diff
@@ -811,8 +816,8 @@ impl ViewClusterBuffers {
 }
 
 // NOTE: With uniform buffer max binding size as 16384 bytes
-// that means we can fit 256 clusterable objects in one uniform
-// buffer, which means the count can be at most 256 so it
+// that means we can fit 204 clusterable objects in one uniform
+// buffer, which means the count can be at most 204 so it
 // needs 9 bits.
```
Comment (eidloi) [inline crates/bevy_pbr/src/cluster/mod.rs:821]: @pcwalton you only need 8 now, right?

**Pull request merged**